Tests/Unit/MSFT_xWebAppPool.Tests.ps1

```diff
@@ -0,0 +1,862 @@
+            if($curVal -eq $value1)
+            {
+              $testParams.Add($option,$value2)
+            }
+            else
+            {
+              $testParams.Add($option,$value1)
+            }
+            
+            Mock Invoke-AppCmd {return $global:PoolCfg}
+            Mock Invoke-AppCmdState {return "Started"}
+            Test-TargetResource @testParams
+            Return 
+        }
+
+        Function SetTargetResource
+        {
+            <#
+
+            .SYNOPSIS
+            Runs Set-TargetResource on $option and then verifies that option was changed to the desired value.  Returns $true if option was changed.
+
+            .DESCRIPTION
+            Runs Set-TargetResource on $option and then verifies that option was changed to the desired value.  Returns $true if option was changed.
+
+            .PARAMETER option 
+            should be the name of the option as defined in xWebAppPool.psm1
+
+            .PARAMETER appcmdVal
+            The location that would be used to get this value via appcmd.
+            
+            .PARAMETER value1
+            A valid value that $option can be set to.  Must be different than value2
+
+            .PARAMETER value2
+            A valid value that $option can be set to.  Must be different than value1
+
+            .EXAMPLE
+            SetTargetResource "autoStart" "add.autoStart" "true" "false"
+
+            #>
+            [OutputType([bool])]
+            Param($option,$appcmdVal,$value1,$value2)
+            $testParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+            $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+            if($curVal -ne $value1)
+            {
+                $testParams.Add($option,$value1)
+                Set-TargetResource @testParams | Out-Null #So we only return the value of Test-TargetResource below
+                $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+                $value1 -eq $curVal
+            }
+            else
+            {
+                $testParams.Add($option,$value2)
+                Set-TargetResource @testParams | Out-Null #So we only return the value of Test-TargetResource below
+                $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+                $value2 -eq $curVal
+            }
+
+            Return 
+        }
+
+        try
+        {
+            $baseParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+
+            It 'DefaultAppPool should exist' {           
```

Comment:
So I have plenty of machines where I've removed the DefaultAppPool as I'm not using it, so why are we making the assumption that it should exist on every machine we're running this code on?